File: library/src/main/java/com/github/easyguide/client/DialogGuideClient.java
package com.github.easyguide.client;

import android.app.Dialog;
import android.view.View;
import android.view.ViewGroup;
import android.widget.PopupWindow;

import com.github.easyguide.EasyGuideManager;

/**
 * Created by shenxl on 2018/8/30.
 */
public class DialogGuideClient extends CommonGuideClient {
    PopupWindow mPopupWindow;
    View mDecorView;

    public DialogGuideClient(EasyGuideManager manager, Dialog dialog) {
        super(manager);
        if (dialog.getWindow() != null) {
            mPopupWindow = new PopupWindow(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
            mPopupWindow.setContentView(manager.getParentView());
            this.mDecorView = dialog.getWindow().getDecorView();
        }
    }

    @Override
    public void showLayer() {
        if (mPopupWindow != null) {
            mPopupWindow.showAsDropDown(mDecorView);
        }
        super.showLayer();
    }

    @Override
    public void dismissCurrent() {
        if (!mManager.hasNextLayer() && mPopupWindow != null) {
            mPopupWindow.dismiss();
        }
        super.dismissCurrent();
    }

    @Override
    public void dismissAll() {
        if (mPopupWindow != null) {
            mPopupWindow.dismiss();
        }
        super.dismissAll();
    }
}

File: library/src/main/java/com/github/easyguide/client/DialogGuideClient.kt
package com.github.easyguide.client

import android.app.Dialog
import android.view.View
import android.view.ViewGroup
import android.widget.PopupWindow

/**
 * Created by shenxl on 2018/8/30.
 */
internal class DialogGuideClient(dialog: Dialog, private var mClient: IGuideClient = CommonGuideClient()) : IGuideClient by mClient{
    private lateinit var popupWindow: PopupWindow
    private var decorView: View = dialog.window?.decorView ?: throw IllegalArgumentException("Window in dialog is null!")

    override fun show() {
        popupWindow = PopupWindow(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT).apply {
            contentView = parentView
            showAsDropDown(decorView)
        }
        mClient.show()
    }

    override fun goNext() {
        if (currentLayer.next == null) {
            popupWindow.dismiss()
        }
        mClient.goNext()
    }

    override fun dismiss() {
        popupWindow.dismiss()
        mClient.dismiss()
    }
}
